app/models/plan.rb

```diff
@@ -391,8 +382,8 @@ def commentable_by?(user_id)
   #
   # Returns Boolean
   def administerable_by?(user_id)
-    user_id = user_id.id if user_id.is_a?(User)
-    role?(user_id, :administrator)
+    role = Role.find_by(plan_id: id, user_id: user_id, active: true)
+    (role.present? ? role.administrator? : false)
```

Comment:
I think we could replace this style query with:
```
Role.administrator.where(plan_id: id, user_id: user_id, active: true),any?
```
In this and the surrounding methods

Reply:
yes, that's cleaner. will change.